app.rb

```diff
@@ -0,0 +1,110 @@
+require_relative 'person'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'classroom'
+require_relative 'book'
+require_relative 'rental'
+
+class App
+  attr_reader :books, :person, :rentals, :classroom
+
+  def initialize
+    @books = []
+    @person = []
+    @rentals = []
+  end
+
+  def list_all_books
+    puts 'All Books:'
+    @books.each do |book|
+      puts "Title: #{book.title}, Author: #{book.author}"
+    end
+  end
+
+  def list_all_people
+    puts 'All People:'
+    @person.each do |person|
+      puts "#{person.class}, Name: #{person.name}, Age: #{person.age}"
+
+      puts "Specialization: [#{person.specialization}]" if person.instance_of?(Teacher)
+    end
+  end
+
+  def create_person
+    puts 'Press 1 for student, 2 for teacher:'
+    is_student = gets.chomp.to_i
+    puts 'Enter person name:'
+    name = gets.chomp
+    puts 'Enter person age:'
+    age = gets.chomp.to_i
+
+    case is_student
+    when 1
+      print 'Does the student have parent permission? [Y/N]: '
+      permission = gets.chomp.downcase == 'y'
+      student = Student.new(name, 1, age, permission)
+      @person << student
+    when 2
+      print 'What is the teacher\'s specialization? '
+      specialization = gets.chomp
+      teacher = Teacher.new(name, specialization, age)
+      @person << teacher
+    else
+      puts 'Invalid input'
+      return
+    end
+
+    puts 'Person created successfully.'
+  end
+
+  def create_book
+    puts 'Enter book title:'
+    title = gets.chomp
+    puts 'Enter book author:'
+    author = gets.chomp
+
+    book = Book.new(title, author)
+    @books << book
+    puts 'Book created successfully.'
+  end
+
+  def create_rental
+    puts 'Select book by number:'
+    @books.each_with_index do |book, index|
+      puts "#{index} - Title: #{book.title}, Author: #{book.author}"
+    end
+
+    book_index = gets.chomp.to_i
+
+    puts 'Select person by number:'
+    @person.each_with_index do |person, index|
+      puts "#{index} - #{person.class}, Name: #{person.name}"
+    end
+
+    person_index = gets.chomp.to_i
+    puts 'Enter date:'
+    date = gets.chomp
+
+    rental = Rental.new(date, @books[book_index], @person[person_index])
+    @rentals << rental
+    puts 'Rental created successfully.'
+  end
+
+  def list_rentals_for_person
+    puts 'All IDs:'
+    @rentals.each do |rental|
+      puts "#{rental.person.id}, Name: #{rental.person.name}"
+    end
+    puts 'Select ID:'
+    id = gets.chomp.to_i
```

Comment:
As it is right now, the code is not working, and if the user tries to return the rental, it will fail.

![image](https://github.com/KAYEMBAJONATHAN/school-library/assets/83596135/8bfbf0e1-30e6-4e36-bb54-b842677e87cd)

But in this code is a small error, because, at the moment, you are transforming the ID into an integer, when it should not be transformed into an integer. Let's just leave it as `id = gets.chomp`.

If we do so, the user will be able to get his rental.
![image](https://github.com/KAYEMBAJONATHAN/school-library/assets/83596135/42dbdeb8-641c-423f-a952-ade2ef56d4da)